internal/watcher/watcher.go

```diff
@@ -80,10 +80,11 @@ func New(ctx context.Context, initialDirs []string, opts ...option) (*Watcher, e
 			case startCmd:
 				// Start from service doesn't pass a context explicitly
 				parentCtx := c.ctx
-				if parentCtx == context.TODO() {
-					parentCtx = ctx
+				if parentCtx == nil {
```

Comment:
Hum, I prefer the previous flow (in the quest of killing as much `else` as possible:
```golang
if parentCtx == nil {
    *parentCtx = ctx
}
ctx, cancel = context.WithCancel(*parentCtx)
```

Reply:
ah, indeed, if parentCtx is nil! :p